attachment_azure/models/ir_attachment.py

```diff
@@ -0,0 +1,165 @@
+        else:
+            try:
+                sas_token = generate_account_sas(
+                    account_name=account_name,
+                    account_key=account_key,
+                    resource_types=ResourceTypes(service=True),
+                    permission=AccountSasPermissions(read=True),
+                    expiry=datetime.utcnow() + timedelta(hours=1),
+                )
+                blob_service_client = BlobServiceClient(
+                    account_url=account_url,
+                    credential=sas_token,
+                )
+            except exceptions.UserError as error:
+                _logger.exception(
+                    "Error during the connection to Azure container using the Shared "
+                    "Access Signature (SAS)"
+                )
+                raise exceptions.UserError(str(error))
+        return blob_service_client
+
+    @api.model
+    def _get_azure_container(self):
+        running_env = os.environ.get("RUNNING_ENV", "dev")
+        container_name = str.lower(running_env + "-" + self.env.cr.dbname)
+        blob_service_client = self._get_blob_service_client()
+        container_client = blob_service_client.get_container_client(container_name)
+        try:
+            # Create the container
+            container_client.create_container()
+        except ResourceExistsError:
+            pass
+        except exceptions.UserError as error:
+            _logger.exception("Error during the creation of the Azure container")
+            raise exceptions.UserError(str(error))
+        return container_client
+
+    @api.model
+    def _store_file_read(self, fname, bin_size=False):
+        if fname.startswith("azure://"):
+            container_client = self._get_azure_container()
+            key = fname.replace("azure://", "", 1).lower()
+            try:
+                blob_client = container_client.get_blob_client(key)
+                read = blob_client.download_blob().readall()
+            except exceptions.UserError:
+                read = ""
+                _logger.info("Attachment '%s' missing on object storage", fname)
+            return read
+        else:
+            return super(IrAttachment, self)._store_file_read(fname, bin_size)
+
+    @api.model
+    def _store_file_write(self, key, bin_data):
+        location = self.env.context.get("storage_location") or self._storage()
+        if location == "azure":
+            container_client = self._get_azure_container()
+            with io.BytesIO() as file:
+                blob_client = container_client.get_blob_client(key.lower())
+                file.write(bin_data)
+                file.seek(0)
+                filename = "azure://%s" % (key)
+                try:
+                    blob_client.upload_blob(file, blob_type="BlockBlob")
+                except ResourceExistsError:
+                    pass
+                except exceptions.UserError as error:
```

Comment:
Is this `except` at the right level? Also seems suspicious, as I doubt `upload_blob` will ever raise such an exception.